src/MOI_wrapper/MOI_wrapper.jl

```diff
@@ -2831,8 +2831,20 @@ function MOI.get(model::Optimizer, attr::MOI.PrimalStatus)
     valueP = Ref{Cint}()
     ret = GRBgetintattr(model, "SolCount", valueP)
     _check_ret(model, ret)
-    return 1 <= attr.result_index <= valueP[] ? MOI.FEASIBLE_POINT :
-           MOI.NO_SOLUTION
+    if !(1 <= attr.result_index <= valueP[])
+        return MOI.NO_SOLUTION
+    end
+    if term in (MOI.OPTIMAL, MOI.SOLUTION_LIMIT)
+        return MOI.FEASIBLE_POINT
+    end
+    # Feasibility of solution is unknown. Check for violations.
+    doubleP = Ref{Cdouble}()
+    ret = GRBgetdblattr(model, "MaxVio", doubleP)
+    _check_ret(model, ret)
+    if doubleP[] < 1e-8
```

Comment:
If we make this `1e-8` a parameter that the user can modify then I think it's fine

Reply:
I don't really like this, because it won't respect any parameters that are set by the user.